<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Read Mapping and Processing Workflow</title>
  <style>
    body {
      font-family: Arial, sans-serif;
      line-height: 1.6;
      margin: 40px;
      background-color: #f9f9f9;
    }
    code, pre {
      background-color: #eee;
      padding: 10px;
      display: block;
      overflow-x: auto;
      white-space: pre-wrap;
      border-left: 5px solid #ccc;
    }
    h2, h3, h4 {
      color: #333;
    }
  </style>
</head>
<body>

  <h2>Task 1: Map Trimmed Reads Using BWA MEM</h2>
  <p><strong>Install BWA</strong></p>
  <pre><code>conda create -n bwa -c bioconda bwa
conda activate bwa</code></pre>

  <p><strong>Map Reads</strong></p>
  <pre><code>bwa mem GCF_018350215.1_P.leo_Ple1_pat1.1_genomic.fna Benin_1_subset_val_1.fastq.gz Benin_2_subset_val_2.fastq.gz > Benin_aligned_reads.sam</code></pre>

  <p><strong>Loop for All Samples</strong></p>
  <pre><code>for file1 in *subset_val_1.fastq.gz; do
  file2=${file1/subset_val_1.fastq.gz/subset_val_2.fastq.gz}
  sample_name=$(basename "$file1" _1_subset_val_1.fastq.gz)
  bwa mem GCF_018350215.1_P.leo_Ple1_pat1.1_genomic.fna "$file1" "$file2" > "${sample_name}_aligned_reads.sam"
done</code></pre>

  <p><strong>Deactivate BWA Environment</strong></p>
  <pre><code>conda deactivate</code></pre>

  <h2>Task 2: Convert SAM to BAM and Sort</h2>
  <p><strong>Install and Activate Samtools</strong></p>
  <pre><code>conda create -n samtools -c bioconda samtools
conda activate samtools</code></pre>

  <p><strong>Convert and Sort</strong></p>
  <pre><code>samtools sort SRR10764405.sam -o SRR10764405_sorted.bam</code></pre>

  <p><strong>Batch Conversion</strong></p>
  <pre><code>for file in *.sam; do 
  samtools sort "$file" -o "${file%.sam}_sorted.bam"
done</code></pre>

  <p><strong>Deactivate Samtools Environment</strong></p>
  <pre><code>conda deactivate</code></pre>

  <h2>Task 3: Mark Duplicates Using GATK4</h2>
  <p><strong>Install and Activate GATK4</strong></p>
  <pre><code>
  conda create -n gatk4 -c bioconda gatk4
  conda activate gatk4
  </code></pre>
  <p><strong>Add readgroup</strong></p>
  <pre><code> 
  for file in *_sorted.bam
  do
  gatk AddOrReplaceReadGroups I=${file} O=${file/_sorted.bam/_sorted_RG.bam} RGID=${file/_sorted.bam/} RGPL=ILLUMINA RGPU=${file/_sorted.bam/} RGSM=${file/_sorted.bam/} RGLB=${file/_sorted.bam/}
  done
  </code></pre>
  

  <p><strong>Mark Duplicates</strong></p>
  <pre><code>gatk MarkDuplicates -I Benin_sorted_reads.bam -O Benin_deduplicated.bam -M Benin_duplication_metrics.txt --REMOVE_DUPLICATES true</code></pre>


  <p><strong>Loop Version</strong></p>
  <pre><code>for file in *_sorted_reads.bam; do
  base=${file%_sorted_reads.bam}
  gatk MarkDuplicates \
    -I "$file" \
    -O "${base}_deduplicated.bam" \
    -M "${base}_duplication_metrics.txt" \
    --REMOVE_DUPLICATES true
done</code></pre>

  <h2>Task 4: Index Deduplicated Files</h2>
  <pre><code>samtools index Benin_deduplicated.bam</code></pre>

  <p><strong>Batch Indexing</strong></p>
  <pre><code>samtools index *_deduplicated.bam


  <h2>Task 5: Estimate Coverage using Qualimap</h2>
  <pre><code>conda create -n qualimap -c bioconda qualimap
conda activate qualimap</code></pre>

  <p><strong>Run on Single File</strong></p>
  <pre><code>qualimap bamqc -bam Benin_deduplicated.bam -outdir qualimap_results -outformat HTML</code></pre>

  <p><strong>Run on All Files</strong></p>
  <pre><code>qualimap bamqc -bam *_deduplicated.bam -outdir qualimap_results -outformat HTML</code></pre>

  <p><strong>Check Output</strong></p>
  <pre><code>cd qualimap_results
cat genome_results.txt</code></pre>

</body>
</html>
